sound/soc/sof/intel/hda-dsp.c

```diff
@@ -696,12 +696,27 @@ int hda_dsp_resume(struct snd_sof_dev *sdev)
 		.state = SOF_DSP_PM_D0,
 		.substate = SOF_HDA_DSP_PM_D0I0,
 	};
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	struct hdac_bus *bus = sof_to_bus(sdev);
+#endif
 	int ret;
 
 	/* resume from D0I3 */
 	if (sdev->dsp_power_state.state == SOF_DSP_PM_D0) {
 		hda_codec_i915_display_power(sdev, true);
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+		/* reenable CORB/RIRB */
+		snd_hdac_bus_init_cmd_io(bus);
+		ret = snd_hdac_ext_bus_link_power_up_all(bus);
+		if (ret < 0) {
+			dev_dbg(sdev->dev,
+				"error %d in %s: failed to power up links",
+				ret, __func__);
+			return ret;
+		}
+#endif
+
```

Comment:
Can you clarify if there is any specific reason why the first thing you do is power-up the link and enable RIRB, before you've changed the DSP power and restored the state?

If it was symmetrical with the suspend routine, you would first restore PCI state and disable wakes *before* re-initializing the links?

Either it's symmetrical by default, or it's not on purpose and you need to document why.

Reply:
@plbossart Could be just that the surrounding existing is not symmetrical either. I.e. we change the i915_display_codec power state before changing DSP state, but order is same for suspend/resume. For i915 this is ok, but for the HDA shutdown in this patch, I think the ordering is right.